resources/git-branch-stage-commit.md

```diff
@@ -0,0 +1,120 @@
+
+Why not work on them in master? The same as above, plus the added risk of making changes directly to the branch that reflects your live app (it could break). 
+
+### Ok cool. How do I create/switch branches?
+Let's say you want to create a new branch for a feature called `favourites`. To **create** this new branch locally and switch to it, use this command:
+```
+$ git checkout -b favourites
+```
+That will give me a message that says:
+```
+Switched to a new branch 'favourites'
+```
+Now to **display a list** of what branches you have locally, use the command `git branch`:
+```
+* favourites
+  master
+```
+The star denotes what branch you are currently on (favourites here). 
+To **switch** back to the master branch, just use the command: 
+```
+$ git checkout master
+```
+That's pretty much it, just replace the branch names in the above commands to whatever branches you are creating or switching to. 
+
+## Staging
+After you've made the changes in your branch, it is time to stage those changes.
+Let's say that in your `favourites` branch, you created a new file called `constants.js` and you modified a file called `index.js`. If you check the state of your files using the command `git status`:
+```
+On branch favourites
+Changes not staged for commit:
+  (use "git add <file>..." to update what will be committed)
+  (use "git checkout -- <file>..." to discard changes in working directory)
+
+	modified:   index.js
+
+Untracked files:
+  (use "git add <file>..." to include in what will be committed)
+
+	constants.js
+
+no changes added to commit (use "git add" and/or "git commit -a")
+```
+Here, `constants.js` is listed as an untracked file because it is new, meaning it was not part of the previous commit, and therefore is not being tracked yet. After this file is included in a commit, it will be tracked. 
+
+Ok, so let's **stage** the changes for both of these files. You can do them individually using the command found in the above message, or do them both using the command: 
+```
+git add .
+```
+Now if you check the state of your files using `git status` again, you'll see this:
+```
+On branch favourites
+Changes to be committed:
+  (use "git reset HEAD <file>..." to unstage)
+
+	modified:   index.js
+	new file:   constants.js
+```
+Now both file changes are staged to be committed. 
+## Committing
+When you make a commit, you essentially take a snapshot of the state of all the files in the directory at that point in time. This is the basis of tracking changes in git. To make a commit, use this command and include a descriptive message of what the commit does:
```

Comment:
I feel like these two lines "When you make a commit, you essentially take a snapshot of the state of all the files in the directory at that point in time. This is the basis of tracking changes in git" are really good (💯) and maybe a version of this can be used at the top to quickly explain what git is for?

Reply:
awesome - yep, will be sure to add them to wherever we decide to put git summary